### ISSUE TITLE
Speed up Rules critical errors tests

### ISSUE THREAD

**Issue by tunetheweb** [sqlfluff/sqlfluff#3034]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

With the recent performance improvement, the CI jobs are taking a lot less time. Except for the "Rules critical errors tests" one which takes 20mins or so.

Could we speed it up? This runs a lint against ALL our dialect fixtures, so is going to take some time, and is a great test but still, it's slowness is annoying and will only grow as we increase out dialect coverage and add new dialects.

I don't think parallel processing will help as it already uses 2 processors, and GitHub Action machines are currently limited to two-core machines anyway.

Could we split it into multiple jobs that can run in parallel? I wouldn't want to hard code the dialects, but maybe have dialects beginning a-m in one job, and n-z in another? Or have ansi (our biggest job) in one job, and the rest in another?

### Use case

PRs not to be held up on a long running job

### Dialect

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
